spylib/admin_api.py

```diff
@@ -187,6 +187,14 @@ async def execute_gql(
             headers=headers,
         )
         jsondata = resp.json()
+        # Handle any response that is not 200, which will return with error message
+        # https://shopify.dev/api/admin-graphql#status_and_error_codes
+        if resp.status_code != 200:
```

Comment:
Some responses may not contain valid JSON - we probably want to do the check before attempting to parse